src/cargo/util/machine_message.rs

```diff
@@ -1,27 +1,24 @@
-use rustc_serialize::Encodable;
-use rustc_serialize::json::{self, Json};
+use serde::ser;
+use serde_json::{self, Value};
 
 use core::{PackageId, Target, Profile};
 
-pub trait Message: Encodable {
+pub trait Message: ser::Serialize {
     fn reason(&self) -> &str;
 }
 
 pub fn emit<T: Message>(t: T) {
-    let json = json::encode(&t).unwrap();
-    let mut map = match json.parse().unwrap() {
-        Json::Object(obj) => obj,
-        _ => panic!("not a json object"),
-    };
-    map.insert("reason".to_string(), Json::String(t.reason().to_string()));
-    println!("{}", Json::Object(map));
+    let mut json: Value = serde_json::to_value(&t).unwrap();
+    json.as_object_mut().unwrap()
+        .insert("reason".to_string(), Value::String(t.reason().to_string()));
```

Comment:
Oh the compile error I encountered was because I forgot to update serde_json.

Ahhh clever! I was flabbergasted this worked but when I checked up on the implementation it was a nice trick!

I approve 👍